components/expenses/EditExpenseDialog.tsx

```diff
@@ -51,8 +54,108 @@ const RenderFormFields = ({ field, onSubmit, expense }) => {
       return <EditAttachments onSubmit={onSubmit} expense={expense} />;
     case 'invoiceFile':
       return <EditInvoiceFile onSubmit={onSubmit} expense={expense} />;
+    case 'paidBy':
+      return <EditPaidBy expense={expense} handleClose={handleClose} />;
   }
 };
+
+const EditPaidBy = ({ expense, handleClose }) => {
+  const schema = z.object({
+    destinationAccount: z.object({
+      id: z.string().optional(),
+      slug: z.string().optional(),
+      name: z.string().optional(),
+    }),
+  });
+
+  const { data, loading } = useQuery(accountsQuery, {
+    variables: {
+      accountSlug: expense.account.parent?.slug ?? expense.account.slug,
+      limit: 100, // TODO: This is the max limit of childrenAccounts, when refactoring the Collective Picker Async to work with GQL v2, this limitation can be worked around
+    },
+    context: API_V2_CONTEXT,
+  });
+
+  const activeAccounts = React.useMemo(
+    () =>
+      data?.account ? [data?.account, ...(data?.account?.childrenAccounts?.nodes.filter(a => a.isActive) || [])] : [],
+    [data?.account],
+  );
+
+  const intl = useIntl();
+  const [moveExpense, { loading: loadingMutation }] = useMutation(moveExpenseMutation, {
```

Comment:
:ice_cream: Personal pref: I like to call `loading` for mutations `submitting`

```suggestion
  const [moveExpense, { loading: submitting }] = useMutation(moveExpenseMutation, {
```

Reply:
That's better!